### PR TITLE
Standarize arguments in function/method calls and instantiation

### PR DESCRIPTION
**Pull request by rragundez**
### Summary
At the moment there is no coding standard when the parameters of a method or instantiating a class are many. This PR standardizes them both. This allows better readability of the code, for example the instantiation of `DataframeIterator`, `DirectoryIterator` and `NumpyArrayIterator` in their respective `flow_from_` methods of `ImageDataGenerator` becomes clearer and easier to track.

### PR Overview

- [ n] This PR requires new unit tests [y/n] (make sure tests are included)
- [n ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y ] This PR is backwards compatible [y/n]
- [n ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by rragundez** [comment]
For example after doing this I just noticed that `flow_from_dataframe` and `DataframeIterator` have a different `x_col` default argument.

**Comment by rragundez** [comment]
@Dref360 I fixed the conflicts

File: keras_preprocessing/image/image_data_generator.py [hunk 1]
```diff
@@ -509,23 +521,35 @@ class subdirectories (default: False).
...
             save_to_dir=save_to_dir,
             save_prefix=save_prefix,
             save_format=save_format,
             follow_links=follow_links,
             subset=subset,
-            interpolation=interpolation)
+            interpolation=interpolation
+        )
 
     def flow_from_dataframe(self, dataframe, directory,
```
Comment (Dref360) [inline keras_preprocessing/image/image_data_generator.py]: not consistent with the rest.

File: keras_preprocessing/image/image_data_generator.py [hunk 2]
```diff
@@ -422,12 +429,17 @@ def flow(self, x,
             save_to_dir=save_to_dir,
             save_prefix=save_prefix,
             save_format=save_format,
-            subset=subset)
+            subset=subset
+        )
 
     def flow_from_directory(self, directory,
```
Comment (Dref360) [inline keras_preprocessing/image/image_data_generator.py]: same

File: keras_preprocessing/image/image_data_generator.py (hunk 1 above)
Comment (rragundez) [reply keras_preprocessing/image/image_data_generator.py]: done

File: keras_preprocessing/image/image_data_generator.py (hunk 2 above)
Comment (rragundez) [reply keras_preprocessing/image/image_data_generator.py]: done

**Comment by rragundez** [comment]
@Dref360 I think this PR is ready to be merged, could you have a look? thanks

**Pull request merged**